parser_library/test/workspace/workspace_pattern_test.cpp

```diff
@@ -0,0 +1,351 @@
+                return pgroups_file_pattern_relative;
+            case variants::URI:
+                return pgroups_file_pattern_uri;
+            case variants::URI_2:
+                return pgroups_file_pattern_uri_2;
+            case variants::ALL_TYPES:
+                return pgroups_file_all_types;
+        }
+        throw std::logic_error("Not implemented");
+    }
+
+public:
+    file_manager_lib_pattern(variants variant)
+    {
+        did_open_file(proc_grps_loc, 1, get_pgroup_file(variant));
+        did_open_file(pgm_conf_loc, 1, pgmconf_file);
+        did_open_file(pattern_test_source_loc, 1, source_txt);
+    }
+
+    MOCK_METHOD(list_directory_result,
+        list_directory_files,
+        (const hlasm_plugin::utils::resource::resource_location& path),
+        (const override));
+
+    list_directory_result list_directory_subdirs_and_symlinks(
+        const hlasm_plugin::utils::resource::resource_location& location) const override
+    {
+        if (location == pattern_test_lib_loc)
+            return { { { pattern_lib_sublib1_abs_path, pattern_test_lib_sublib1_loc },
+                         { pattern_lib_sublib2_abs_path, pattern_test_lib_sublib2_loc } },
+                hlasm_plugin::utils::path::list_directory_rc::done };
+
+        if (location == user_dir_loc)
+            return { { { "pattern_est", pattern_est_dir_loc },
+                         { "pattern_test", pattern_test_dir_loc },
+                         { "patter_test", patter_test_dir_loc } },
+                hlasm_plugin::utils::path::list_directory_rc::done };
+
+        if (location == pattern_test_dir_loc)
+            return { { { "libs", pattern_test_lib_loc } }, hlasm_plugin::utils::path::list_directory_rc::done };
+
+
+        if (location == different_libs2_libs_loc)
+            return different_libs2_libs_subdir;
+
+        if (location == temp_lib2_libs_loc)
+            return temp_lib2_libs_subdir;
+
+        return { {}, hlasm_plugin::utils::path::list_directory_rc::done };
+    }
+
+    bool dir_exists(const hlasm_plugin::utils::resource::resource_location&) const override { return true; }
+};
+} // namespace
+
+TEST(workspace_pattern_test, absolute)
+{
+    file_manager_lib_pattern file_manager(variants::ABSOLUTE);
+    lib_config config;
+
+    workspace ws(ws_loc, "workspace_name", file_manager, config);
+    ws.open();
+
+    EXPECT_CALL(file_manager, list_directory_files(temp_lib_loc))
+        .WillOnce(::testing::Return(list_directory_result { {}, hlasm_plugin::utils::path::list_directory_rc::done }));
+    EXPECT_CALL(file_manager, list_directory_files(temp_lib2_libs_loc))
+        .WillOnce(::testing::Return(list_directory_result { {}, hlasm_plugin::utils::path::list_directory_rc::done }));
+    EXPECT_CALL(file_manager, list_directory_files(temp_lib2_libs_subdir.first.begin()->second))
+        .WillOnce(::testing::Return(list_directory_result { {}, hlasm_plugin::utils::path::list_directory_rc::done }));
```

Comment:
Is this order dependent? The `list_directory_result` type contains an unordered_map, so it could cause unexpected issues...